Files/Helpers/ExtensionManager.cs

```diff
@@ -0,0 +1,531 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.ApplicationModel;
+using Windows.ApplicationModel.AppExtensions;
+using Windows.ApplicationModel.AppService;
+using Windows.Foundation.Collections;
+using Windows.Storage;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Media.Imaging;
+
+namespace Files.Helpers
+{
+    class ExtensionManager
+    {
+        //private CoreDispatcher _dispatcher; // used to run code on the UI thread for code that may update UI
+        private AppExtensionCatalog _catalog; // the catalog of app extensions available to this host
+
+        /// <summary>
+        /// Builds a collection of extensions available to this host
+        /// </summary>
+        /// <param name="extensionContractName">
+        /// The contract string is defined in the extension host's Package.appxmanifest file under <uap3:AppExtensionHost><uap3:Name>MathExt</uap3:Name></uap3:AppExtensionHost> 
+        /// The string is defined in the extension's Package.appxmanifest file under <uap3:Extension Category="windows.appExtension"><uap3:AppExtension Name = "MathExt" ... >
+        /// When these two strings match, the extension is loaded.</param>
+        public ExtensionManager(string extensionContractName)
+        {
+            // catalog & contract
+            ExtensionContractName = extensionContractName;
+            _catalog = AppExtensionCatalog.Open(ExtensionContractName);
+            //_dispatcher = null;
+        }
+
+        // The collection of extensions for this host
+        public ObservableCollection<Extension> Extensions { get; } = new ObservableCollection<Extension>();
+
+        // new jtw
+        public Extension GetExtension(string id)
+        {
+            return Extensions.Where(e => e.UniqueId == id).FirstOrDefault();
+        }
+
+        // The name that extensions must specify to be considered an extension this host can load
+        public string ExtensionContractName { get; private set; }
+
+        /// <summary>
+        /// Sets up handlers for package events
+        /// Be sure to call this from the UI thread.
+        /// </summary>
+        /// <param name="dispatcher"></param>
+        public void Initialize()
+        {
+            #region Error Checking & Dispatcher Setup
+            // verify that we haven't already been initialized
+            //if (_dispatcher != null)
```

Comment:
Remove unnecesary code

Reply:
@d2dyno1 I think this code should be kept as it's for updating the UI (say, a page for managing extensions), but isn't currently implemented.